Add new template for BLC-connected seed-ui

vigneshranganathan-eaton opened the pull request:
<!-- Include a bulleted list summarizing the main changes you have made in this PR -->

#### Changes proposed in this Pull Request:

- New feature Seed UI template which contains Admin invite, Admin user invite registration process, Self Invite, Self invite registration process. 

<!-- Include screenshots if they will help illustrate the changes in this PR -->

#### Screenshots / Screen Recording (if applicable)

-
![image](https://github.com/etn-ccis/blui-react-cli-templates/assets/116724074/d21145aa-70a0-4613-867c-e94a150d2fdb)


<!-- Instruction for PR reviewers, if more complicated than a simple yarn start -->

#### To Test:

- Enter "npm run build:local"
- Select React in project framework
- Provide application name
- Select Seed UI in template
- Select Eslint true
- Select Prettier true
- npm start

<!-- Useful for draft pull requests -->

#### Any specific feedback you are looking for?

-

huayunh commented:
made an issue, BLUI-4281, to track the effort

JeffGreiner-eaton commented:
I was trying to test the seedui-typescript file using local copy of blui cli.

It's been a while since I tested this but I'm in the cli-templates repo locally running `blui new ` from there using  
`blui new --template=file:./blui-react-cli-templates/seedui-typescript`

 

I throw this error for any template  ---any ideas?

<img width="1285" alt="image" src="https://github.com/etn-ccis/blui-react-cli-templates/assets/119693939/80067d3a-0430-4be5-874b-83301c823094">

joebochill commented:
I'm having some issues getting my local environment to run the CLI so I asked @daileytj to have a look at this.

daileytj commented:
> I was trying to test the seedui-typescript file using local copy of blui cli.
> 
> It's been a while since I tested this but I'm in the cli-templates repo locally running `blui new ` from there using  `blui new --template=file:./blui-react-cli-templates/seedui-typescript`
> 
>  
> 
> I throw this error for any template  ---any ideas?
> 
> <img alt="image" width="1285" src="https://user-images.githubusercontent.com/119693939/240688684-80067d3a-0430-4be5-874b-83301c823094.png">

try like this @JeffGreiner-eaton ... no `=` sign next to template
`blui new --template file:./blui-react-cli-templates/seedui-typescript`

daileytj commented:
@vigneshranganathan-eaton can you update the dependencies to include the `@brightlayer-ui/...` and `@mui/...` packages used for this template.

You can follow this [react getting started guide](https://brightlayer-ui.github.io/development/frameworks-web/react) for all your dependencies.

Aside from that I was able to get the project to generate, but not build. I cannot access the `@etn-electrical/derms-blcloud-seedui` package with my token via npm, which I'm not sure is the right way to install since this is a GitHub package. How should I install this package as it doesn't look like it's in the dependencies either?

JeffGreiner-eaton commented:
> 

ok yes, will need to update the doc in CLI repo...
https://github.com/etn-ccis/blui-cli#testing-templates

vigneshranganathan-eaton commented:
> 



> @vigneshranganathan-eaton can you update the dependencies to include the `@brightlayer-ui/...` and `@mui/...` packages used for this template.
> 
> You can follow this [react getting started guide](https://brightlayer-ui.github.io/development/frameworks-web/react) for all your dependencies.
> 
> Aside from that I was able to get the project to generate, but not build. I cannot access the `@etn-electrical/derms-blcloud-seedui` package with my token via npm, which I'm not sure is the right way to install since this is a GitHub package. How should I install this package as it doesn't look like it's in the dependencies either?

@daileytj I have already added '@brightlayer-ui/ and @mui/' packages in template.json as a dependency
![image](https://github.com/etn-ccis/blui-react-cli-templates/assets/116724074/dcbc38f2-a277-4177-aac4-721c954f91a4)
I also faced same issue in local while testing, so i have moved all the dependency into devdependency for testing purpose then it worked for me.

vigneshranganathan-eaton commented on seedui-typescript/template/.npmrc:
```diff
@@ -0,0 +1,3 @@
+//npm.pkg.github.com/:_authToken=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
```
@daileytj  I assumed my token would be sufficient for the package installation so only i pushed. I will resolve that.
Any suggestion from your side how to make it work, either i need to make it has publicrepo or publish the package similar to blui in npm.

joebochill commented:
After getting my token set up correctly to access the package, I was able to run everything after going through the CLI using a local file template. But I don't know how to log in...is there a test account that you're using? I tried to self register, but I never got the email.

vigneshranganathan-eaton commented:
> After getting my token set up correctly to access the package, I was able to run everything after going through the CLI using a local file template. But I don't know how to log in...is there a test account that you're using? I tried to self register, but I never got the email.

vigneshranganathan-eaton commented:
> After getting my token set up correctly to access the package, I was able to run everything after going through the CLI using a local file template. But I don't know how to log in...is there a test account that you're using? I tried to self register, but I never got the email.

@joebochill  I have dropped mail with config information and sample login credential, you can use that credential and validation.
